x/ccv/provider/keeper/key_assignment.go

```diff
@@ -370,13 +370,22 @@ func (k Keeper) DeleteConsumerAddrsToPrune(ctx sdk.Context, chainID string, vscI
 }
 
 // AssignConsumerKey assigns the consumerKey to the validator with providerAddr
-// on the consumer chain with ID chainID
+// on the consumer chain with ID chainID, if it is either registered or currently
+// voted on in a ConsumerAddition governance proposal
 func (k Keeper) AssignConsumerKey(
 	ctx sdk.Context,
 	chainID string,
 	validator stakingtypes.Validator,
 	consumerKey tmprotocrypto.PublicKey,
 ) error {
+	// check that the consumer chain is either registered or that
+	// ConsumerAdditionProposal was voted on.
+	if !k.CheckIfConsumerIsProposedOrRegistered(ctx, chainID) {
```

Comment:
@mpoke This is the change that prevents key assignments for non-registered consumer chains.